addok/helpers/collectors.py

```diff
@@ -102,3 +105,56 @@ def sorter(t):
             helper.add_to_bucket(keys)
             if helper.bucket_overflow:
                 break
+
+
+def extend_results_extrapoling_relations(helper):
+    if not helper.bucket_dry:
+        return  # No need.
+    tokens = set(helper.meaningful + helper.common)
+    for relation in _extract_manytomany_relations(tokens):
+        helper.add_to_bucket([t.db_key for t in relation])
+        if helper.bucket_overflow:
+            break
+
+
+def _extract_manytomany_relations(tokens):
```

Comment:
I'm not sure this method is still useful, what about integrating it within `extend_results_extrapolating_relations` directly?